src/client.rs

```diff
@@ -19,8 +21,34 @@ use tokio_service::Service;
 type WireResponse<Resp, E> = Result<Result<Resp, WireError<E>>, DeserializeError>;
 type ResponseFuture<Req, Resp, E> = futures::Map<<BindClient<Req, Resp, E> as Service>::Future,
                                             fn(WireResponse<Resp, E>) -> Result<Resp, ::Error<E>>>;
-type BindClient<Req, Resp, E> =
-    <Proto<Req, Result<Resp, WireError<E>>> as ProtoBindClient<Multiplex, TcpStream>>::BindClient;
+type BindClient<Req, Resp, E> = <Proto<Req, Result<Resp, WireError<E>>> as
+                                ProtoBindClient<Multiplex, StreamType>>::BindClient;
+
+/// TLS-specific functionality
+#[cfg(feature = "tls")]
+pub mod tls {
+    use native_tls::TlsConnector;
+
+    /// TLS context
+    pub struct TlsClientContext {
```

Comment:
Yeah, I actually was considering that as well. I didn't particularly like putting `Client` in the name, but thought it would be useful if it was available at the root namespace; `::TlsContext` wouldn't be very descriptive.

I like the idea of a root `tls` module.